rtlib/rope.hh

```diff
@@ -405,8 +435,57 @@ class Ref {
         append(s);
     }
 
-    template <typename O>
-    void put(O &o) const {
+    void put(std::ofstream &o) const {
+      if (readonly == true) {
+        Block<Refcount>* i = first;
+        while (i) {
```

Comment:
when does `i` evaluate to false?

Reply:
`i` is `false` when it equals `NULL` or `nullptr`.